web/channels/player_channel.ex

```diff
@@ -0,0 +1,38 @@
+defmodule Yggdrasil.PlayerChannel do
+  use Yggdrasil.Web, :channel
+  alias Yggdrasil.Player
+  alias Yggdrasil.Endpoint
+
+  def join("player:" <> user_id = topic, _message, socket) do
+    if socket.assigns.user == String.to_integer(user_id) do
+      socket = assign socket, :player_topic, topic
+      {:ok, socket}
+    else
+      {:error, %{ error: "auth failure"} }
+    end
+  end
+
+  def handle_in("join_game", _message, socket) do
+    player_topic = socket.assigns.player_topic
+    push_msg = fn (payload) ->
+      Endpoint.broadcast! player_topic, "event", payload
+    end
+
+    case Player.join_game socket.assigns.user, self, push_msg do
```

Comment:
Ah I see yeah.

At some point this might be needed or something similar

* user joins game lobby
* user is told to join player channel, the game is made aware of this
* user joins player channel and game verifies it was expecting this user

Maybe we expand the game channel to handle different events like "join_game", the user sends this event to do what you said about selecting the character, and then given a player channel topic, the user then joins that topic and the game verifies it was issued a token to play. At that point we could probably drop user_id and use like UUID or something so that it's unique and generated by the game saying I am authorizing you to join this session.

Not something needed for now though I have feeling this stuff will shake out as we test and find ways to get around what is been written.